src/go/k8s/cmd/main.go

```diff
@@ -352,6 +361,13 @@ func main() {
 	case OperatorV2Mode:
 		ctrl.Log.Info("running in v2", "mode", OperatorV2Mode, "helm controllers enabled", enableHelmControllers, "namespace", namespace)
 
+		factory, err := internalclient.NewClientFactory(mgr.GetConfig())
+		if err != nil {
+			setupLog.Error(err, "Unable to create client factory")
+			os.Exit(1)
+		}
+		factory = factory.WithLogger(mgr.GetLogger().WithName("ClientFactory"))
```

Comment:
So, one thing about the change to a shared factory pattern like this is that we don't necessarily have a per-request scoped logger for the factory. I generally wanted to keep the internal logging we were doing in the topic controller more or less the same and didn't want to overhaul the logging done in the internal `KafkaAPISpec` client initialization... but, this change does mean that we're no longer seeing the full request logging context of a log line in the factory calls.

If that's a problem for debugging I might consider changing `WithLogger` to making a full copy of the factory and returning it so that we can call it per-request and pass in the request-scoped logger each time in the reconciler loop. Thoughts?

Reply:
Could we just use `context.Context` to pass along the logger? There's plenty of convenience methods for doing so in the `log` package and some examples of using them in the pvcunbinder. Then there's no need to expose the internal detail of logging at all, if I'm understanding you correctly.

Reply:
That's a really good idea. I'm thinking that internally I could just pull the logger out of the context, and if we can't because it's not in the context, then maybe just default to the `Discard` logger.

Reply:
That or fall back to the globally set logger (which is the default behavior of the `log` package)